spec/controllers/news_controller_spec.rb

```diff
@@ -0,0 +1,281 @@
+require "spec_helper"
+
+RSpec.describe NewsController do
+  describe "GET index" do
+    it "works logged out" do
+      get :index
+      expect(response).to have_http_status(200)
+    end
+
+    it "works logged in" do
+      login
+      get :index
+      expect(response).to have_http_status(200)
+    end
+
+    it "only shows one news post" do
+      n1 = create(:news)
+      n2 = create(:news)
+      get :index
+      expect(assigns(:news).to_a).to eq([n2])
+      expect(assigns(:news).total_pages).to eq(2)
+    end
+
+    it "marks the news post read" do
+      user = create(:user)
+      news = create(:news)
+      expect(NewsView.find_by(user: user)).to be_nil
+
+      login_as(user)
+      get :index
+
+      expect(NewsView.find_by(user: user).news).to eq(news)
+    end
+  end
+
+  describe "GET new" do
+    it "errors if logged out" do
+      get :new
+      expect(response).to redirect_to(root_url)
+      expect(flash[:error]).to eq("You must be logged in to view that page.")
+    end
+
+    it "errors if not staff" do
+      login
+      get :new
+      expect(response).to redirect_to(news_index_url)
+      expect(flash[:error]).to eq("You do not have permission to manage news posts.")
+    end
+
+    it "works for staff" do
+      login_as(create(:mod_user))
+      get :new
+      expect(response).to have_http_status(200)
+      expect(assigns(:page_title)).to eq('Create News Post')
+    end
+  end
+
+  describe "POST create" do
+    it "errors if logged out" do
+      post :create
+      expect(response).to redirect_to(root_url)
+      expect(flash[:error]).to eq("You must be logged in to view that page.")
+    end
+
+    it "errors if not staff" do
+      login
+      post :create
+      expect(response).to redirect_to(news_index_url)
+      expect(flash[:error]).to eq("You do not have permission to manage news posts.")
+    end
+
+    it "errors without content" do
+      login_as(create(:mod_user))
+      post :create, params: {news: {content: ''}}
+      expect(response).to render_template(:new)
+      expect(assigns(:page_title)).to eq('Create News Post')
+      expect(flash[:error][:message]).to eq("News post could not be created.")
+    end
+
+    it "works for staff" do
+      login_as(create(:mod_user))
+      expect {
+        post :create, params: {news: {content: 'staff made this'}}
+      }.to change { News.count }.by(1)
+      expect(response).to redirect_to(news_index_url)
+      expect(News.last.content).to eq('staff made this')
+    end
+  end
+
+  describe "GET show" do
+    it "errors without news post" do
```

Comment:
I think this is in the wrong section (PATCH update)

Reply:
I think it's a bad copy paste not the wrong section but yes